src/prompt_constructor.py

```diff
@@ -301,24 +353,36 @@ def prompt_generate_custom_cuda_from_file_one_example(ref_arch_src, example_ind=
     example_arch = read_file(example_arch_path)
     example_new_arch = read_file(example_new_arch_path)
 
-    return prompt_generate_custom_cuda(arch, example_arch, example_new_arch)
+    return prompt_generate_custom_kernel(arch, example_arch, example_new_arch)
 
 
-def prompt_generate_custom_cuda_from_prompt_template(ref_arch_src: str) -> str:
+def prompt_generate_custom_kernel_from_prompt_template(
+    ref_arch_src: str, framework: str = "cuda"
+) -> str:
     """
     Using prompt example (an element-wise addition) for prompt templates
     The most basic form of example just to show LLM the task and the expected output format
     """
+
+    assert framework in ["cuda", "triton"], "Framework must be either cuda or triton"
+
     arch = ref_arch_src
     # These are strictly defined for now
 
     # path to prompt template, show an example of Model (torch specifications) and ModelNew (torch + custom CUDA kernels)
-    example_arch_path = os.path.join(
-        REPO_TOP_PATH, f"src/prompts/model_ex_add.py"
-    )
-    example_new_arch_path = os.path.join(
-        REPO_TOP_PATH, f"src/prompts/model_new_ex_add.py"
-    )
+    example_arch_path = os.path.join(REPO_TOP_PATH, f"src/prompts/model_ex_add.py")
+    if framework == "cuda":
+        example_new_arch_path = os.path.join(
+            REPO_TOP_PATH, f"src/prompts/model_new_ex_add.py"
```

Comment:
os.path.join is OS agnostic so would recommend instead syntax like `os.path.join(..., "src", "prompts")`

Also this string should not be an f-string